### ISSUE TITLE
Factorio Compatibility 

### ISSUE THREAD

**Issue by arturh85** [panicbit/rust-rcon#13]
I'm trying to use rust-rcon to send commands to a Factorio server instead of a Minecraft server.
It works but loops forever instead of returning.

The issue is the following bit of code in lib.rs:70:

```rust
        self.send(PacketType::ExecCommand, cmd).await?;
...
        // the server processes packets in order, so send an empty packet and
        // remember its id to detect the end of a multi-packet response
        let end_id = self.send(PacketType::ExecCommand, "").await?;
        let mut result = String::new();

        loop {
            let received_packet = self.recv().await?;

            if received_packet.get_id() == end_id {
                // This is the response to the end-marker packet
                break;
            }

            result += received_packet.get_body();
        }
```

If I add logging I see that end_id is 3 the the only package ever received is 2. I guess Factorio ignores the empty send and only returns confirmation for the original send. I fixed the issue by removing the original end_id line and instead use the real send call:

```rust
let end_id = self.send(PacketType::ExecCommand, cmd).await?;
```

Could you please add this difference in behaviour as an additional feature toggle?

**Comment by panicbit** [comment]
I can try looking into this tomorrow. Can you provide me an example series of commands I can check this with?

**Comment by arturh85** [comment]
Sure, you can start Factorio with 

```
factorio --start-server saves/level.zip --rcon-port 1234 --rcon-password test
```

Otherwise it won't open the rcon port.

Then you can try it with the following example program:

```
use rcon::{Connection, Error};

#[tokio::main]
async fn main() -> Result<(), Error> {
    let address = "localhost:1234";
    let mut conn = Connection::connect(address, "test").await?;

    demo(&mut conn, "print('hello world')").await?;
    println!("command finished");
    Ok(())
}

async fn demo(conn: &mut Connection, cmd: &str) -> Result<(), Error> {
    println!("request: {}", cmd);
    let resp = conn.cmd(cmd).await?;
    println!("response: {}", resp);
    Ok(())
}
```

This prints out 
```
request: print('hello world')
```

then loops forever. The Factorio server logs:

```
 487.816 Info RemoteCommandProcessor.cpp:242: New RCON connection from IP ADDR:({127.0.0.1:37807})
2020-07-25 16:44:06 [CHAT] <server>: print('hello world')
```

**Comment by panicbit** [comment]
I'm not sure if I understood your solution incorrectly, but I think it would have sent the command twice?
At any rate you have correctly assessed that the Factorio server doesn't reply to empty packets, which prompts me to assume for now that multi-packet responses are not implemented in Factorio, at least not in a common way.
As a result I added a quirks mode for Factorio that only tries to receive single packets.
Take note that I moved away from using feature flags in favor of an API that allows them to be configured at runtime:
```rust
let mut conn = Connection::builder()
    .enable_factorio_quirks(true)
    .connect(address, "hunter2").await?;
```
I've released this as `0.3.0` on **crates.io**.

Last but not least, **THANK YOU** for this detailed issue and the detailed example! It made working on this in my limited spare time that much easier. :partying_face:
